### DISCUSSION TITLE
PiHole not working?

### DISCUSSION THREAD

**Discussion by LogicalUnit** [StreamController/StreamController#116] [Q&A] [answered]
Hi everyone, I have a pihole on my network at 192.168.1.39. I can access the admin page as expected and it is definitely working. I would love to have a streamdeck button to temporarily disable the pihole. I have installed the pihole plugin and configured the IP address and password. Nothing seems to happen in the pihole GUI when I press the button (the green Active light doesn't change to red). Any suggestions?
NOTE: I am not actually running this on RPi hardware, but rather on a Debian 11 lxc container.

**Comment by Core447** [comment]
Hi,

can you please share a screenshot of the action settings? And can you please close the app completely (via the option in the hamburger menu) and start it from the console (`flatpak run com.core447.StreamController`) to check if you're getting any errors when pressing the key?

**Reply by LogicalUnit** [reply]
@Core447 Here are the configuration screenshot and terminal output. When I press the button on the StreamDeck the icon animates, but there is no indication of any other activity.

![Screenshot from 2024-05-26 13-58-44](https://github.com/StreamController/StreamController/assets/6601496/82785597-ad4f-4ec3-b082-93155101bde4)

![Screenshot from 2024-05-26 14-01-27](https://github.com/StreamController/StreamController/assets/6601496/4152baea-232e-4620-91d3-6f9b7804144a)

**Reply by Core447** [reply]
Are you sure the pihole hasn't been disabled? The icon should always show the fetched state, so it shouldn't update if the disable command didn't work. Have you tried reloading the hole website after pressing the button? If this doesn't work, can you please add two disable actions and see if the second one updates when you press the other one?

**Reply by LogicalUnit** [reply]
@Core447 The buttons for the Disable and Toggle Active actions on the StreamDeck+ always display a slashed RPi icon. When I push the button, the icon momentarily gets smaller, but the slash never goes away. I am certain the PiHole is online and responding to queries (I have checked the query log).

**Reply by Core447** [reply, marked as answer]
Ah ok, sorry, I thought the icon would change. However, I may have found the problem: The input row is labelled "Password", but it requires the api key of your pi-hole. You can get it by going to Settings -> API -> Show api token. If this was the problem, I'll change the label.

**Reply by LogicalUnit** [reply]
Yep, that did the trick. The moment I put the API key into StreamController I saw the nice red RPi icon (before it was grey), and the disable function worked exactly as expected (had to refresh the RPi web page to see the updated state). Cheers!
